examples/firebase-auth-firestore/app/routes/logout.tsx

```diff
@@ -0,0 +1,21 @@
+import { Form, LoaderFunction, redirect } from "remix";
+import { destroySession, getSession } from "~/sessions";
+
+export const action: LoaderFunction = async ({ request }) => {
+  const session = await getSession(request.headers.get("Cookie"));
+  return redirect("/", {
+    headers: { "Set-Cookie": await destroySession(session) },
+  });
+};
+
+export default function Logout() {
+  return (
+    <div>
+      <h1>Logout</h1>
+      <p>Press the button below to log out.</p>
+      <Form method="post">
+        <button type="submit">Logout</button>
+      </Form>
+    </div>
```

Comment:
Is this Logout component required? I get an application error without it in my Firebase implementation, but I'm trying to follow the jokes tutorial as best practice.

https://remix.run/docs/en/v1/tutorials/jokes#build-logout-action

Reply:
I put the form in the same route for clarity. It should also be possible to have the form in a different route, just like in the jokes example. Can you share your implementation?

I thought about building the jokes tutorial with Firebase as an example, but the [contributing guidelines for examples](https://github.com/remix-run/remix/tree/main/examples#contributing) recommend minimalistic examples. The jokes example should show how to use Remix in a real-world application, while this example should only show how to use Firebase with Remix.

Reply:
You did a stellar job. I finally traced my issues down to [Module Constraints](https://remix.run/docs/en/v1/guides/constraints). The conventions you have followed using `.server.ts` and `app/sessions.ts` make perfect sense and fit the remix recommendations. 

I was able to successfully merge the Jokes example with this example. The hardest part was that before this conversation I was chasing seemly cryptic application error messages. I can post the combined version if you want to see it.

Thanks for the help!